torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1024,6 +1024,47 @@ void ProcessGroupNCCL::performNocolorSplit(at::Device device) {
 #endif
 }
 
+void ProcessGroupNCCL::registerUserBuffers(at::Device device) {
+  auto active_pool = c10::cuda::MemPoolContext::getActiveMemPool();
+  TORCH_INTERNAL_ASSERT(
+      active_pool,
+      "registerUserBuffers expects a user provided pool that is currently active");
+
+  const auto key = getKeyFromDevice(device);
+  LOG(INFO) << logPrefix()
+            << "Performing user buffer registration on backend device "
+            << device << ", key " << key << ", i am " << this;
+  auto ncclComm = getNCCLComm(key, device, OpType::ALLREDUCE);
```

Comment:
Do you mind leaving a `HACK` here? 
For enabling NVLS, using `OpType::ALLREDUCE` is fine. I don't know if we would reuse the same API to register buffers for zero-copy P2P. If we do, at least we know there is a hack that needs change.

Reply:
Hack is not needed anymore since `getNCCLComm` can now get the communicator without `OpType`.

Reply:
nvm, `OpType::ALLREDUCE` is still needed to initialize comms. Left the note about `HACK`.